{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>Storage.save in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1 allows directory traversal if crafted filenames are directly passed to it.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-45452\">https://nvd.nist.gov/vuln/detail/CVE-2021-45452</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1\">https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1</a></li>\n<li><a href=\"https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b\">https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b</a></li>\n<li><a href=\"https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6\">https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220121-0005\">https://security.netapp.com/advisory/ntap-20220121-0005</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/jan/04/security-releases\">https://www.djangoproject.com/weblog/2022/jan/04/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jrh2-hc4r-7jwx\">https://github.com/advisories/GHSA-jrh2-hc4r-7jwx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx",
            "title": "[Django] Directory-traversal in Django",
            "date_modified": "2024-04-03T15:00:48.000Z",
            "date_published": "2022-01-12T19:21:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Storage.save in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1 allows directory traversal if crafted filenames are directly passed to it.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-45452\">https://nvd.nist.gov/vuln/detail/CVE-2021-45452</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1\">https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1</a></li>\n<li><a href=\"https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b\">https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b</a></li>\n<li><a href=\"https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6\">https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220121-0005\">https://security.netapp.com/advisory/ntap-20220121-0005</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/jan/04/security-releases\">https://www.djangoproject.com/weblog/2022/jan/04/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jrh2-hc4r-7jwx\">https://github.com/advisories/GHSA-jrh2-hc4r-7jwx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx",
            "title": "[Django] Directory-traversal in Django",
            "date_modified": "2024-04-03T15:00:48.000Z",
            "date_published": "2022-01-12T19:21:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Storage.save in Django 2.2 before 2.2.26, 3.2 before 3.2.11, and 4.0 before 4.0.1 allows directory traversal if crafted filenames are directly passed to it.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2021-45452\">https://nvd.nist.gov/vuln/detail/CVE-2021-45452</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1\">https://github.com/django/django/commit/4cb35b384ceef52123fc66411a73c36a706825e1</a></li>\n<li><a href=\"https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b\">https://github.com/django/django/commit/8d2f7cff76200cbd2337b2cf1707e383eb1fb54b</a></li>\n<li><a href=\"https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6\">https://github.com/django/django/commit/e1592e0f26302e79856cc7f2218ae848ae19b0f6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220121-0005\">https://security.netapp.com/advisory/ntap-20220121-0005</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/jan/04/security-releases\">https://www.djangoproject.com/weblog/2022/jan/04/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-jrh2-hc4r-7jwx\">https://github.com/advisories/GHSA-jrh2-hc4r-7jwx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx",
            "title": "[Django] Directory-traversal in Django",
            "date_modified": "2024-04-03T15:00:48.000Z",
            "date_published": "2022-01-12T19:21:04.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-jrh2-hc4r-7jwx"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Django 3.2 before 3.2.19, 4.x before 4.1.9, and 4.2 before 4.2.1, it was possible to bypass validation when using one form field to upload multiple files. This multiple upload has never been supported by forms.FileField or forms.ImageField (only the last uploaded file was validated). However, Django's \"Uploading multiple files\" documentation suggested otherwise.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-31047\">https://nvd.nist.gov/vuln/detail/CVE-2023-31047</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd\">https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd</a></li>\n<li><a href=\"https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64\">https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64</a></li>\n<li><a href=\"https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965\">https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.2/releases/security\">https://docs.djangoproject.com/en/4.2/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230609-0008\">https://security.netapp.com/advisory/ntap-20230609-0008</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/may/03/security-releases\">https://www.djangoproject.com/weblog/2023/may/03/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r3xc-prgr-mg9p\">https://github.com/advisories/GHSA-r3xc-prgr-mg9p</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p",
            "title": "[Django] Django bypasses validation when using one form field to upload multiple files",
            "date_modified": "2024-04-03T14:59:16.000Z",
            "date_published": "2023-05-07T03:30:17.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Django 3.2 before 3.2.19, 4.x before 4.1.9, and 4.2 before 4.2.1, it was possible to bypass validation when using one form field to upload multiple files. This multiple upload has never been supported by forms.FileField or forms.ImageField (only the last uploaded file was validated). However, Django's \"Uploading multiple files\" documentation suggested otherwise.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-31047\">https://nvd.nist.gov/vuln/detail/CVE-2023-31047</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd\">https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd</a></li>\n<li><a href=\"https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64\">https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64</a></li>\n<li><a href=\"https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965\">https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.2/releases/security\">https://docs.djangoproject.com/en/4.2/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230609-0008\">https://security.netapp.com/advisory/ntap-20230609-0008</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/may/03/security-releases\">https://www.djangoproject.com/weblog/2023/may/03/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r3xc-prgr-mg9p\">https://github.com/advisories/GHSA-r3xc-prgr-mg9p</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p",
            "title": "[Django] Django bypasses validation when using one form field to upload multiple files",
            "date_modified": "2024-04-03T14:59:16.000Z",
            "date_published": "2023-05-07T03:30:17.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Django 3.2 before 3.2.19, 4.x before 4.1.9, and 4.2 before 4.2.1, it was possible to bypass validation when using one form field to upload multiple files. This multiple upload has never been supported by forms.FileField or forms.ImageField (only the last uploaded file was validated). However, Django's \"Uploading multiple files\" documentation suggested otherwise.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-31047\">https://nvd.nist.gov/vuln/detail/CVE-2023-31047</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd\">https://github.com/django/django/commit/21b1b1fc03e5f9e9f8c977ee6e35618dd3b353dd</a></li>\n<li><a href=\"https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64\">https://github.com/django/django/commit/e7c3a2ccc3a562328600be05068ed9149e12ce64</a></li>\n<li><a href=\"https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965\">https://github.com/django/django/commit/eed53d0011622e70b936e203005f0e6f4ac48965</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/django/PYSEC-2023-61.yaml</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.2/releases/security\">https://docs.djangoproject.com/en/4.2/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/A45VKTUVQ2BN6D5ZLZGCM774R6QGFOHW</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/DNEHD6N435OE2XUFGDAAVAXSYWLCUBFD</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230609-0008\">https://security.netapp.com/advisory/ntap-20230609-0008</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/may/03/security-releases\">https://www.djangoproject.com/weblog/2023/may/03/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r3xc-prgr-mg9p\">https://github.com/advisories/GHSA-r3xc-prgr-mg9p</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p",
            "title": "[Django] Django bypasses validation when using one form field to upload multiple files",
            "date_modified": "2024-04-03T14:59:16.000Z",
            "date_published": "2023-05-07T03:30:17.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r3xc-prgr-mg9p"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in the Multipart Request Parser in Django 3.2 before 3.2.18, 4.0 before 4.0.10, and 4.1 before 4.1.7. Passing certain inputs (e.g., an excessive number of parts) to multipart forms could result in too many open files or memory exhaustion, and provided a potential vector for a denial-of-service attack.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-24580\">https://nvd.nist.gov/vuln/detail/CVE-2023-24580</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2023/02/14/1\">http://www.openwall.com/lists/oss-security/2023/02/14/1</a></li>\n<li><a href=\"https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92\">https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92</a></li>\n<li><a href=\"https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432\">https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432</a></li>\n<li><a href=\"https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8\">https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html\">https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.1/releases/security\">https://docs.djangoproject.com/en/4.1/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230316-0006\">https://security.netapp.com/advisory/ntap-20230316-0006</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/feb/14/security-releases\">https://www.djangoproject.com/weblog/2023/feb/14/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2hrw-hx67-34x6\">https://github.com/advisories/GHSA-2hrw-hx67-34x6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6",
            "title": "[Django] Resource exhaustion in Django",
            "date_modified": "2024-04-03T14:58:09.000Z",
            "date_published": "2023-02-15T03:30:47.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in the Multipart Request Parser in Django 3.2 before 3.2.18, 4.0 before 4.0.10, and 4.1 before 4.1.7. Passing certain inputs (e.g., an excessive number of parts) to multipart forms could result in too many open files or memory exhaustion, and provided a potential vector for a denial-of-service attack.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-24580\">https://nvd.nist.gov/vuln/detail/CVE-2023-24580</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2023/02/14/1\">http://www.openwall.com/lists/oss-security/2023/02/14/1</a></li>\n<li><a href=\"https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92\">https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92</a></li>\n<li><a href=\"https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432\">https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432</a></li>\n<li><a href=\"https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8\">https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html\">https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.1/releases/security\">https://docs.djangoproject.com/en/4.1/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230316-0006\">https://security.netapp.com/advisory/ntap-20230316-0006</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/feb/14/security-releases\">https://www.djangoproject.com/weblog/2023/feb/14/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2hrw-hx67-34x6\">https://github.com/advisories/GHSA-2hrw-hx67-34x6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6",
            "title": "[Django] Resource exhaustion in Django",
            "date_modified": "2024-04-03T14:58:09.000Z",
            "date_published": "2023-02-15T03:30:47.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in the Multipart Request Parser in Django 3.2 before 3.2.18, 4.0 before 4.0.10, and 4.1 before 4.1.7. Passing certain inputs (e.g., an excessive number of parts) to multipart forms could result in too many open files or memory exhaustion, and provided a potential vector for a denial-of-service attack.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-24580\">https://nvd.nist.gov/vuln/detail/CVE-2023-24580</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2023/02/14/1\">http://www.openwall.com/lists/oss-security/2023/02/14/1</a></li>\n<li><a href=\"https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92\">https://github.com/django/django/commit/628b33a854a9c68ec8a0c51f382f304a0044ec92</a></li>\n<li><a href=\"https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432\">https://github.com/django/django/commit/83f1ea83e4553e211c1c5a0dfc197b66d4e50432</a></li>\n<li><a href=\"https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8\">https://github.com/django/django/commit/a665ed5179f5bbd3db95ce67286d0192eff041d8</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html\">https://lists.debian.org/debian-lts-announce/2023/02/msg00023.html</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.1/releases/security\">https://docs.djangoproject.com/en/4.1/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKYVMMR7RPM6AHJ2SBVM2LO6D3NGFY7B</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HWY6DQWRVBALV73BPUVBXC3QIYUM24IK</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LTZVAKU5ALQWOKFTPISE257VCVIYGFQI</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VZS4G6NSZWPTVXMMZHJOJVQEPL3QTO77</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YJB6FUBBLVKKG655UMTLQNN6UQ6EDLSP</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20230316-0006\">https://security.netapp.com/advisory/ntap-20230316-0006</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2023/feb/14/security-releases\">https://www.djangoproject.com/weblog/2023/feb/14/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2hrw-hx67-34x6\">https://github.com/advisories/GHSA-2hrw-hx67-34x6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6",
            "title": "[Django] Resource exhaustion in Django",
            "date_modified": "2024-04-03T14:58:09.000Z",
            "date_published": "2023-02-15T03:30:47.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2hrw-hx67-34x6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in MultiPartParser in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2. Passing certain inputs to multipart forms could result in an infinite loop when parsing files.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-23833\">https://nvd.nist.gov/vuln/detail/CVE-2022-23833</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a\">https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a</a></li>\n<li><a href=\"https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468\">https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6cw3-g6wv-c2xv\">https://github.com/advisories/GHSA-6cw3-g6wv-c2xv</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv",
            "title": "[Django] Infinite Loop in Django",
            "date_modified": "2024-04-03T14:55:56.000Z",
            "date_published": "2022-02-04T00:00:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in MultiPartParser in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2. Passing certain inputs to multipart forms could result in an infinite loop when parsing files.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-23833\">https://nvd.nist.gov/vuln/detail/CVE-2022-23833</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a\">https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a</a></li>\n<li><a href=\"https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468\">https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6cw3-g6wv-c2xv\">https://github.com/advisories/GHSA-6cw3-g6wv-c2xv</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv",
            "title": "[Django] Infinite Loop in Django",
            "date_modified": "2024-04-03T14:55:56.000Z",
            "date_published": "2022-02-04T00:00:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in MultiPartParser in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2. Passing certain inputs to multipart forms could result in an infinite loop when parsing files.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-23833\">https://nvd.nist.gov/vuln/detail/CVE-2022-23833</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a\">https://github.com/django/django/commit/c477b761804984c932704554ad35f78a2e230c6a</a></li>\n<li><a href=\"https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468\">https://github.com/django/django/commit/d16133568ef9c9b42cb7a08bdf9ff3feec2e5468</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6cw3-g6wv-c2xv\">https://github.com/advisories/GHSA-6cw3-g6wv-c2xv</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv",
            "title": "[Django] Infinite Loop in Django",
            "date_modified": "2024-04-03T14:55:56.000Z",
            "date_published": "2022-02-04T00:00:26.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6cw3-g6wv-c2xv"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-04-03T14:54:36.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-04-03T14:54:36.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The <code>{% debug %}</code> template tag in Django 2.2 before 2.2.27, 3.2 before 3.2.12, and 4.0 before 4.0.2 does not properly encode the current context. This may lead to XSS.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-22818\">https://nvd.nist.gov/vuln/detail/CVE-2022-22818</a></li>\n<li><a href=\"https://groups.google.com/forum/#!forum/django-announce\">https://groups.google.com/forum/#!forum/django-announce</a></li>\n<li><a href=\"https://www.debian.org/security/2022/dsa-5254\">https://www.debian.org/security/2022/dsa-5254</a></li>\n<li><a href=\"https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5\">https://github.com/django/django/commit/01422046065d2b51f8f613409cad2c81b39487e5</a></li>\n<li><a href=\"https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2\">https://github.com/django/django/commit/1a1e8278c46418bde24c86a65443b0674bae65e2</a></li>\n<li><a href=\"https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6\">https://github.com/django/django/commit/c27a7eb9f40b64990398978152e62b6ff839c2e6</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/4.0/releases/security\">https://docs.djangoproject.com/en/4.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B4SQG2EAF4WCI2SLRL6XRDJ3RPK3ZRDV</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20220221-0003\">https://security.netapp.com/advisory/ntap-20220221-0003</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2022/feb/01/security-releases\">https://www.djangoproject.com/weblog/2022/feb/01/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-95rw-fx8r-36v6\">https://github.com/advisories/GHSA-95rw-fx8r-36v6</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6",
            "title": "[django] Cross-site Scripting in Django",
            "date_modified": "2024-04-03T14:54:36.000Z",
            "date_published": "2022-02-04T00:00:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-95rw-fx8r-36v6"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2024-28219\">https://nvd.nist.gov/vuln/detail/CVE-2024-28219</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security\">https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/2a93aba5cfcf6e241ab4f9392c13e3b74032c061\">https://github.com/python-pillow/Pillow/commit/2a93aba5cfcf6e241ab4f9392c13e3b74032c061</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-44wm-f244-xhp3\">https://github.com/advisories/GHSA-44wm-f244-xhp3</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-44wm-f244-xhp3",
            "title": "[pillow] Pillow buffer overflow vulnerability",
            "date_modified": "2024-04-03T14:53:39.000Z",
            "date_published": "2024-04-03T03:30:30.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-44wm-f244-xhp3"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in Django version 2.2 before 2.2.13 and 3.0 before 3.0.7. Query parameters generated by the Django admin ForeignKeyRawIdWidget were not properly URL encoded, leading to a possibility of an XSS attack.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-13596\">https://nvd.nist.gov/vuln/detail/CVE-2020-13596</a></li>\n<li><a href=\"https://groups.google.com/forum/#!msg/django-announce/pPEmb2ot4Fo/X-SMalYSBAAJ\">https://groups.google.com/forum/#!msg/django-announce/pPEmb2ot4Fo/X-SMalYSBAAJ</a></li>\n<li><a href=\"https://www.debian.org/security/2020/dsa-4705\">https://www.debian.org/security/2020/dsa-4705</a></li>\n<li><a href=\"https://www.oracle.com/security-alerts/cpujan2021.html\">https://www.oracle.com/security-alerts/cpujan2021.html</a></li>\n<li><a href=\"https://github.com/django/django/commit/1f2dd37f6fcefdd10ed44cb233b2e62b520afb38\">https://github.com/django/django/commit/1f2dd37f6fcefdd10ed44cb233b2e62b520afb38</a></li>\n<li><a href=\"https://github.com/django/django/commit/6d61860b22875f358fac83d903dc629897934815\">https://github.com/django/django/commit/6d61860b22875f358fac83d903dc629897934815</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/3.0/releases/security\">https://docs.djangoproject.com/en/3.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4A2AP4T7RKPBCLTI2NNQG3T6MINDUUMZ\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4A2AP4T7RKPBCLTI2NNQG3T6MINDUUMZ</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20200611-0002\">https://security.netapp.com/advisory/ntap-20200611-0002</a></li>\n<li><a href=\"https://usn.ubuntu.com/4381-1\">https://usn.ubuntu.com/4381-1</a></li>\n<li><a href=\"https://usn.ubuntu.com/4381-2\">https://usn.ubuntu.com/4381-2</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2020/jun/03/security-releases\">https://www.djangoproject.com/weblog/2020/jun/03/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2m34-jcjv-45xf\">https://github.com/advisories/GHSA-2m34-jcjv-45xf</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2m34-jcjv-45xf",
            "title": "[django] XSS in Django",
            "date_modified": "2024-04-03T14:50:59.000Z",
            "date_published": "2020-06-05T16:24:28.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2m34-jcjv-45xf"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in Django version 2.2 before 2.2.13 and 3.0 before 3.0.7. Query parameters generated by the Django admin ForeignKeyRawIdWidget were not properly URL encoded, leading to a possibility of an XSS attack.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-13596\">https://nvd.nist.gov/vuln/detail/CVE-2020-13596</a></li>\n<li><a href=\"https://groups.google.com/forum/#!msg/django-announce/pPEmb2ot4Fo/X-SMalYSBAAJ\">https://groups.google.com/forum/#!msg/django-announce/pPEmb2ot4Fo/X-SMalYSBAAJ</a></li>\n<li><a href=\"https://www.debian.org/security/2020/dsa-4705\">https://www.debian.org/security/2020/dsa-4705</a></li>\n<li><a href=\"https://www.oracle.com/security-alerts/cpujan2021.html\">https://www.oracle.com/security-alerts/cpujan2021.html</a></li>\n<li><a href=\"https://github.com/django/django/commit/1f2dd37f6fcefdd10ed44cb233b2e62b520afb38\">https://github.com/django/django/commit/1f2dd37f6fcefdd10ed44cb233b2e62b520afb38</a></li>\n<li><a href=\"https://github.com/django/django/commit/6d61860b22875f358fac83d903dc629897934815\">https://github.com/django/django/commit/6d61860b22875f358fac83d903dc629897934815</a></li>\n<li><a href=\"https://docs.djangoproject.com/en/3.0/releases/security\">https://docs.djangoproject.com/en/3.0/releases/security</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4A2AP4T7RKPBCLTI2NNQG3T6MINDUUMZ\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4A2AP4T7RKPBCLTI2NNQG3T6MINDUUMZ</a></li>\n<li><a href=\"https://security.netapp.com/advisory/ntap-20200611-0002\">https://security.netapp.com/advisory/ntap-20200611-0002</a></li>\n<li><a href=\"https://usn.ubuntu.com/4381-1\">https://usn.ubuntu.com/4381-1</a></li>\n<li><a href=\"https://usn.ubuntu.com/4381-2\">https://usn.ubuntu.com/4381-2</a></li>\n<li><a href=\"https://www.djangoproject.com/weblog/2020/jun/03/security-releases\">https://www.djangoproject.com/weblog/2020/jun/03/security-releases</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2m34-jcjv-45xf\">https://github.com/advisories/GHSA-2m34-jcjv-45xf</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2m34-jcjv-45xf",
            "title": "[django] XSS in Django",
            "date_modified": "2024-04-03T14:50:59.000Z",
            "date_published": "2020-06-05T16:24:28.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2m34-jcjv-45xf"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>Any deployment of voilà dashboard allow local file inclusion, that is to say any file on a filesystem that is readable by the user that runs the voilà dashboard server can be downloaded by someone with network access to the server. </p>\n<p>Whether this still requires authentication depends on how voilà is deployed.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This is patched in 0.2.17+, 0.3.8+, 0.4.4+, 0.5.6+</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>None.</p>\n<h3 id=\"references\">References</h3>\n<p>CWE-73: External Control of File Name or Path</p>\n<h3 id=\"original-report\">Original report</h3>\n<p>I have found a local file inclusion vulnerability in one of your subprojects, voila (<a href=\"https://github.com/voila-dashboards/voila\">https://github.com/voila-dashboards/voila</a>).</p>\n<p>The vulnerability exists in the \"/static\" Route, and can be exploited by simply making a request such as this:</p>\n<pre><code>$ curl localhost:8866/static/etc/passwd\n</code></pre>\n<p>...or by using a webbrowser to download the file.</p>\n<p>I dug into the source code, and I think the offending line is here: <a href=\"https://github.com/voila-dashboards/voila/blob/8419cc7d79c0bb1dabfbd9ec49cb957740609d4d/voila/app.py#L664\">https://github.com/voila-dashboards/voila/blob/8419cc7d79c0bb1dabfbd9ec49cb957740609d4d/voila/app.py#L664</a>\n<code>\"static_path\"</code> gets set to <code>\"/\"</code>, irrespective of the actual <code>\"--static\"</code> cli option. Because of that, the <code>tornado.web.StaticFileHandler</code> gets initialized with <code>path=\"/\"</code>. Then, <code>tornado.web.StaticFileHandler.get</code> calls <code>tornado.web.StaticFileHandler.get_absolute_path</code> with <code>root=\"/\"</code> and <code>path=\"[USER SUPPLIED PATH]\"</code>, which leads to local file inclusion. An attacker can request any file on the system they want (that the user running voila has access to).</p>\n<p>I suspect this was an oversight during development. Setting <code>static_path=self.static_root</code> (the aforementioned correct cli option) in line 664 provides the intended behavior and restricts the file access to the static directory.\nFrom what I can tell, this line has been in the repository since September 2018. This is the commit that added it: <a href=\"https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67\">https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67</a></p>\n<p>I have found multiple voila instances online that are impacted, such as:</p>\n<ul>\n<li>... [redacted]</li>\n<li>... [redacted]</li>\n<li>... [redacted]</li>\n</ul>\n<p>...but many more probably exist. They're easy to identify by <code>[redacted]</code> Therefore the Issue should be fixed as soon as possible, and a security advisory should be released to inform the impacted users.</p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/voila-dashboards/voila/security/advisories/GHSA-2q59-h24c-w6fg\">https://github.com/voila-dashboards/voila/security/advisories/GHSA-2q59-h24c-w6fg</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/00d6362c237b6b4d466873535554d6076ead0c52\">https://github.com/voila-dashboards/voila/commit/00d6362c237b6b4d466873535554d6076ead0c52</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67\">https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/5542e4ae36bb5d184deaa48f95e76be477756af2\">https://github.com/voila-dashboards/voila/commit/5542e4ae36bb5d184deaa48f95e76be477756af2</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/98b6a40fec27723572314fdbba99bdc147d904c8\">https://github.com/voila-dashboards/voila/commit/98b6a40fec27723572314fdbba99bdc147d904c8</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/c045be6988539d07cceeb9f82fc660a49485d504\">https://github.com/voila-dashboards/voila/commit/c045be6988539d07cceeb9f82fc660a49485d504</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2q59-h24c-w6fg\">https://github.com/advisories/GHSA-2q59-h24c-w6fg</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2q59-h24c-w6fg",
            "title": "[voila] Voilà Local file inclusion",
            "date_modified": "2024-04-03T14:13:10.000Z",
            "date_published": "2024-04-03T14:13:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2q59-h24c-w6fg"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>Any deployment of voilà dashboard allow local file inclusion, that is to say any file on a filesystem that is readable by the user that runs the voilà dashboard server can be downloaded by someone with network access to the server. </p>\n<p>Whether this still requires authentication depends on how voilà is deployed.</p>\n<h3 id=\"patches\">Patches</h3>\n<p>This is patched in 0.2.17+, 0.3.8+, 0.4.4+, 0.5.6+</p>\n<h3 id=\"workarounds\">Workarounds</h3>\n<p>None.</p>\n<h3 id=\"references\">References</h3>\n<p>CWE-73: External Control of File Name or Path</p>\n<h3 id=\"original-report\">Original report</h3>\n<p>I have found a local file inclusion vulnerability in one of your subprojects, voila (<a href=\"https://github.com/voila-dashboards/voila\">https://github.com/voila-dashboards/voila</a>).</p>\n<p>The vulnerability exists in the \"/static\" Route, and can be exploited by simply making a request such as this:</p>\n<pre><code>$ curl localhost:8866/static/etc/passwd\n</code></pre>\n<p>...or by using a webbrowser to download the file.</p>\n<p>I dug into the source code, and I think the offending line is here: <a href=\"https://github.com/voila-dashboards/voila/blob/8419cc7d79c0bb1dabfbd9ec49cb957740609d4d/voila/app.py#L664\">https://github.com/voila-dashboards/voila/blob/8419cc7d79c0bb1dabfbd9ec49cb957740609d4d/voila/app.py#L664</a>\n<code>\"static_path\"</code> gets set to <code>\"/\"</code>, irrespective of the actual <code>\"--static\"</code> cli option. Because of that, the <code>tornado.web.StaticFileHandler</code> gets initialized with <code>path=\"/\"</code>. Then, <code>tornado.web.StaticFileHandler.get</code> calls <code>tornado.web.StaticFileHandler.get_absolute_path</code> with <code>root=\"/\"</code> and <code>path=\"[USER SUPPLIED PATH]\"</code>, which leads to local file inclusion. An attacker can request any file on the system they want (that the user running voila has access to).</p>\n<p>I suspect this was an oversight during development. Setting <code>static_path=self.static_root</code> (the aforementioned correct cli option) in line 664 provides the intended behavior and restricts the file access to the static directory.\nFrom what I can tell, this line has been in the repository since September 2018. This is the commit that added it: <a href=\"https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67\">https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67</a></p>\n<p>I have found multiple voila instances online that are impacted, such as:</p>\n<ul>\n<li>... [redacted]</li>\n<li>... [redacted]</li>\n<li>... [redacted]</li>\n</ul>\n<p>...but many more probably exist. They're easy to identify by <code>[redacted]</code> Therefore the Issue should be fixed as soon as possible, and a security advisory should be released to inform the impacted users.</p>\n<h3 id=\"references-1\">References</h3>\n<ul>\n<li><a href=\"https://github.com/voila-dashboards/voila/security/advisories/GHSA-2q59-h24c-w6fg\">https://github.com/voila-dashboards/voila/security/advisories/GHSA-2q59-h24c-w6fg</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/00d6362c237b6b4d466873535554d6076ead0c52\">https://github.com/voila-dashboards/voila/commit/00d6362c237b6b4d466873535554d6076ead0c52</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67\">https://github.com/voila-dashboards/voila/commit/28faacc9b03b160fd8fa920ad045f4ec0667ab67</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/5542e4ae36bb5d184deaa48f95e76be477756af2\">https://github.com/voila-dashboards/voila/commit/5542e4ae36bb5d184deaa48f95e76be477756af2</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/98b6a40fec27723572314fdbba99bdc147d904c8\">https://github.com/voila-dashboards/voila/commit/98b6a40fec27723572314fdbba99bdc147d904c8</a></li>\n<li><a href=\"https://github.com/voila-dashboards/voila/commit/c045be6988539d07cceeb9f82fc660a49485d504\">https://github.com/voila-dashboards/voila/commit/c045be6988539d07cceeb9f82fc660a49485d504</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-2q59-h24c-w6fg\">https://github.com/advisories/GHSA-2q59-h24c-w6fg</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-2q59-h24c-w6fg",
            "title": "[voila] Voilà Local file inclusion",
            "date_modified": "2024-04-03T14:13:10.000Z",
            "date_published": "2024-04-03T14:13:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-2q59-h24c-w6fg"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}